language/diem-framework/modules/NetworkIdentity.move

```diff
@@ -0,0 +1,143 @@
+/// Module managing Diemnet NetworkIdentity
+module DiemFramework::NetworkIdentity {
+    use DiemFramework::DiemTimestamp;
+    use Std::Errors;
+    use Std::Event::{Self, EventHandle};
+    use Std::Signer;
+    use Std::Vector;
+
+    /// Holder for all `NetworkIdentity` in an account
+    struct NetworkIdentity has key {
+        identities: vector<vector<u8>>,
+        /// Event handle for `identities` rotation events
+        identity_change_events: EventHandle<NetworkIdentityChangeNotification>
+    }
+
+    /// Message sent when there are updates to the `NetworkIdentity`.
+    struct NetworkIdentityChangeNotification has drop, store {
+        /// The new identities
+        identities: vector<vector<u8>>,
+        /// The time at which the `identities` was rotated
+        time_rotated_seconds: u64,
+    }
+
+    const MAX_ADDR_IDENTITIES: u64 = 100;
+
+    // Error Codes
+    /// Limit exceeded on number of identities for an address
+    const ENET_ID_LIMIT_EXCEEDED: u64 = 0;
+
+    // =================================================================
+    // Module Specification
+
+    spec module {} // Switch to module documentation context
+
+    /// Initialize `NetworkIdentity` with an empty list
+    fun initialize_network_identity(account: &signer) {
+        let identities = Vector::empty<vector<u8>>();
+        let identity_change_events = Event::new_event_handle<NetworkIdentityChangeNotification>(account);
+        move_to(account, NetworkIdentity { identities, identity_change_events });
+    }
+    spec initialize_network_identity {
+        pragma opaque;
+        let account_addr = Signer::spec_address_of(account);
+        ensures exists<NetworkIdentity>(account_addr);
+        modifies global<NetworkIdentity>(account_addr);
+    }
+
+    /// Return the underlying `NetworkIdentity` bytes
+    public fun get(account_addr: address): vector<vector<u8>> acquires NetworkIdentity {
+        *&borrow_global<NetworkIdentity>(account_addr).identities
```

Comment:
In the Diem Framework we try to not have any `borrow_globals` without first checking for the existence of the resource (makes it _much_ easier to debug an issue). Could you add a new error code (e.g., `ENETWORK_ID_DOESNT_EXIST`) and add an assertion to check the existence of this resource before calling borrow_global here?
```suggestion
         assert(exists<NetworkIdentity>(account_addr), Errors::not_published(ENETWORK_ID_DOESNT_EXIST));
        *&borrow_global<NetworkIdentity>(account_addr).identities
```